integration-tests/helpers/index.js

```diff
@@ -189,6 +189,46 @@ function spawnProc (filename, options = {}, stdioHandler, stderrHandler) {
   })
 }
 
+async function execHelper (command, options) {
+  if (command.startsWith('yarn')) {
+    try {
+      console.log('Commencing exec execution...', command) // eslint-disable-line no-console
+      await execWithTimeout(command, options)
+      console.log('Success it ran!') // eslint-disable-line no-console
+    } catch (e) {
+      console.error('We caught, commencing retry, error:', // eslint-disable-line no-console
+        e, 'failed command:', command)
+      try {
+        await execWithTimeout(command, options)
+        console.log('Success on the second try!') // eslint-disable-line no-console
+      } catch {
+        console.error('Retry failed, command:', command) // eslint-disable-line no-console
+      }
+    }
+  } else {
+    console.log('Commencing exec execution without retrying:', command) // eslint-disable-line no-console
+    await exec(command, options)
+  }
+}
+
+function execWithTimeout (command, options, timeoutMs = 5000) {
+  const execPromise = exec(command, options)
+  const timeoutPromise = new Promise((resolve, reject) =>
+    setTimeout(() => reject(new Error(`Timeout after ${timeoutMs}ms`)), timeoutMs)
+  )
+
+  return Promise.allSettled([execPromise, timeoutPromise]).then(results => {
+    const [execResult, timeoutResult] = results
+
+    if (execResult.status === 'rejected') {
+      throw execResult.reason
+    }
+
+    if (timeoutResult.status === 'rejected') {
+      throw timeoutResult.reason
+    }
+  })
```

Comment:
I am not sure if I understand this correct.

If I read the code correct, it will wait for both promises to be settled, no matter how long the command takes and if the timeout already triggered. If it was longer than the timeout and the command succeeded, the command will still be considered failed due to the timeout promise being rejected. Thus, I don't understand why it would not be a `Promise.race`, since I don't understand in what way it is helpful to wait longer until we log the timeout.